src/software/thunderscope/gl/helpers/gl_patches.py

```diff
@@ -66,6 +69,27 @@ def GLViewMixin_removeItem_patched(self, item: GLGraphicsItem) -> None:
     self.update()
 
 
+def ShaderProgram_program_patched(original: Callable) -> Callable:
+    """Returns a patched version of ShaderProgram.program that forces
+    recompilation of the shader program when the OpenGL context changes.
```

Comment:
We should probably add a note that says that one `ShaderProgram` object should never be used across multiple contexts simultaneously (i.e. the same shader being used in two drawings in different windows), since that would now result in unnecessary compilation instead of being a hard error(?)